mp/src/game/client/momentum/ui/SettingsPanel/OnlineSettingsPage.cpp

```diff
@@ -21,62 +21,30 @@ OnlineSettingsPage::OnlineSettingsPage(Panel* pParent) : BaseClass(pParent, "Onl
     m_pEnableGhostTrails->AddActionSignalTarget(this);
     m_pEnableColorAlphaOverride = new CvarToggleCheckButton(this, "EnableAlphaOverride", "#MOM_Settings_Override_Alpha_Enable", "mom_ghost_online_alpha_override_enable");
     m_pEnableColorAlphaOverride->AddActionSignalTarget(this);
-    m_pAlphaOverrideSlider = new CvarSlider(this, "AlphaOverrideSlider", nullptr, 0.0f, 255.0f, "mom_ghost_online_alpha_override");
+    m_pAlphaOverrideSlider = new CvarSlider(this, "AlphaOverrideSlider", nullptr, 0.0f, 255.0f, "mom_ghost_online_alpha_override", false, true);
     m_pAlphaOverrideSlider->AddActionSignalTarget(this);
-    m_pAlphaOverrideInput = new TextEntry(this, "AlphaOverrideEntry");
+    m_pAlphaOverrideInput = new CvarTextEntry(this, "AlphaOverrideEntry", "mom_ghost_online_alpha_override");
     m_pAlphaOverrideInput->SetAllowNumericInputOnly(true);
     m_pAlphaOverrideInput->AddActionSignalTarget(this);
 
     LoadControlSettings("resource/ui/SettingsPanel_OnlineSettings.res");
 }
 
-void OnlineSettingsPage::OnApplyChanges()
-{
-    BaseClass::OnApplyChanges();
-
-    
-}
-
 void OnlineSettingsPage::LoadSettings()
 {
-
-    UpdateSliderSettings();
-}
-
-void OnlineSettingsPage::OnCheckboxChecked(Panel* p)
-{
-}
-
-void OnlineSettingsPage::OnTextChanged(vgui::Panel* panel)
-{
-    BaseClass::OnTextChanged(panel);
-
-    if (panel == m_pAlphaOverrideInput)
+    if (m_pEnableColorAlphaOverride && m_pAlphaOverrideSlider && m_pAlphaOverrideInput)
     {
-        char buf[64];
-        m_pAlphaOverrideInput->GetText(buf, 64);
-
-        float fValue = static_cast<float>(Q_atof(buf));
-        if (fValue >= 0.0f && fValue <= 255.0f)
-        {
-            m_pAlphaOverrideSlider->SetSliderValue(fValue);
-        }
+        m_pAlphaOverrideSlider->SetEnabled(m_pEnableColorAlphaOverride->IsSelected());
+        m_pAlphaOverrideInput->SetEnabled(m_pEnableColorAlphaOverride->IsSelected());
     }
 }
 
-void OnlineSettingsPage::OnControlModified(vgui::Panel* p)
+void OnlineSettingsPage::OnCheckboxChecked(Panel *p)
 {
-    BaseClass::OnControlModified(p);
-
-    if (p == m_pAlphaOverrideSlider)
+    BaseClass::OnCheckboxChecked(p);
+    if (p == m_pEnableColorAlphaOverride && m_pAlphaOverrideSlider && m_pAlphaOverrideInput)
```

Comment:
No need to null check these, they should be valid

Reply:
Found that this was the case on the old `GameplaySettingsPage.cpp` code as well. I fixuped my commit on relating to the `OnlineSettingsPage` and added a new commit for removing this on the `GameplaySettingsPage`.